In [5]:
# Enter your information below
Student.Name = 'Laurelyne Barbier'
Student.Number = 67493288
sprintf('GEOS 300 Assignment Submission for %s: %s',Student.Name,Student.Number)

[1] "GEOS 300 Assignment Submission for Laurelyne Barbier: 67493288"

In [6]:
install.packages("latex2exp")
library("tidyverse")
library("ggplot2")
library("reshape2")
library("latex2exp")
library("patchwork")
library("tidyr")
library("knitr")
library("plyr")
library("lubridate")
library("zoo")
library("readr")

Updating HTML index of packages in '.Library'

Making 'packages.html' ...
 done

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.4.2     ✔ purrr   1.0.1
✔ tibble  3.2.1     ✔ dplyr   1.1.1
✔ tidyr   1.3.0     ✔ stringr 1.5.0
✔ readr   2.1.3     ✔ forcats 0.5.2
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()

Attaching package: ‘reshape2’


The following object is masked from ‘package:tidyr’:

    smiths


------------------------------------------------------------------------------

You have loaded plyr after dplyr - this is likely to cause problems.
If you need functions from both plyr and dplyr, please load plyr first, then dplyr:
library(plyr); library(dplyr)

------------------------------------------------------------------------------


Attaching package: ‘plyr’


The following objects are masked from ‘package:dplyr’:

    arr

## Question 1

Download your data and print descriptive statistics for all columns. How do maximum and minimum air temperature at 0.4 m compare to those at 2.0 m?

In [7]:
data_url='https://raw.githubusercontent.com/GEOS300/AssignmentData/main/CA_DBB_DailyData/20160820.txt'
df <- read.csv(file = data_url,na.strings="-9999")

df$TIMESTAMP <- as.POSIXct(df$Date.Time..PST.,format = "%Y-%m-%d %H:%M:%S")

head(df)
summary(subset(df,select = -c(TIMESTAMP, Date.Time..PST.)))

Date.Time..PST.     T_0.4m..K. T_2m..K. u_0.4m..m.s.1. u_2m..m.s.1.
1 2016-08-20 00:30:00 284.47     287.65   0.00           0.24        
2 2016-08-20 01:00:00 284.22     286.93   0.20           0.38        
3 2016-08-20 01:30:00 283.30     285.94   0.07           0.25        
4 2016-08-20 02:00:00 283.04     285.35   0.28           0.43        
5 2016-08-20 02:30:00 282.56     284.64   0.00           0.11        
6 2016-08-20 03:00:00 282.17     284.38   0.00           0.24        
  Rn..W.m.2. u.w...m2.s.2. w.T...K.m.s.1. w.rho_v...g.m.2.s.1.
1 -20.20     -0.0128       -0.0114        -0.0016             
2 -18.13     -0.0057        0.0060         0.0027             
3 -16.09     -0.0048        0.0029         0.0022             
4 -15.31     -0.0040        0.0047         0.0024             
5 -13.29     -0.0011       -0.0021         0.0006             
6 -13.87     -0.0186        0.0060         0.0018             
  w.rho_CO2...micromol.m.2.s.1. w.rho_CH4...microåmol.m.2.s.1. Pa..kPa.
1 2.0350                        0.1379                         101.260 
2 2.0026                        0.1353                         101.255 
3 1.9728                        0.1329                         101.250 
4 1.9337                        0.1298                         101.255 
5 1.8984                        0.1271                         101.260 
6 1.8647                        0.1246                         101.255 
  PPFD..micromol.m.2.s.1. rv         TIMESTAMP          
1 101.260                 0.01449635 2016-08-20 00:30:00
2 101.255                 0.01379902 2016-08-20 01:00:00
3 101.250                 0.01309974 2016-08-20 01:30:00
4 101.255                 0.01266678 2016-08-20 02:00:00
5 101.260                 0.01209807 2016-08-20 02:30:00
6 101.255                 0.01192066 2016-08-20 03:00:00

   T_0.4m..K.       T_2m..K.     u_0.4m..m.s.1.    u_2m..m.s.1.   
 Min.   :282.0   Min.   :284.0   Min.   :0.0000   Min.   :0.0600  
 1st Qu.:285.6   1st Qu.:287.5   1st Qu.:0.1250   1st Qu.:0.2725  
 Median :294.6   Median :294.0   Median :0.4800   Median :1.1800  
 Mean   :293.6   Mean   :293.2   Mean   :0.6992   Mean   :1.1579  
 3rd Qu.:301.0   3rd Qu.:299.0   3rd Qu.:1.2825   3rd Qu.:1.9975  
 Max.   :305.4   Max.   :301.4   Max.   :1.5400   Max.   :2.5000  
   Rn..W.m.2.     u.w...m2.s.2.       w.T...K.m.s.1.      w.rho_v...g.m.2.s.1.
 Min.   :-63.13   Min.   :-0.091700   Min.   :-0.013400   Min.   :-0.003000   
 1st Qu.:-20.36   1st Qu.:-0.055475   1st Qu.:-0.000925   1st Qu.: 0.000675   
 Median : 22.89   Median :-0.017000   Median : 0.010800   Median : 0.009750   
 Mean   :171.29   Mean   :-0.031598   Mean   : 0.052331   Mean   : 0.025848   
 3rd Qu.:404.40   3rd Qu.:-0.004675   3rd Qu.: 0.130525   3rd Qu.: 0.051800   
 Max.   :590.67   Max.   :-0.000600   Max.   : 0.176900  

At 0.4m, the minimum temperature is 282.0 K while the maximum temperature is 305.4 K. The minimum and maximum temperatures at 2m are 284.0 K and 301.4 K respectively. The minimum temperature is thus lower while the maximum temperature is higher at 0.4m. The air at 2m experiences a narrower range of temperatures than the air at 0.4m, by a few degrees.

## Question 2

Convert the air temperature values at 0.4 m and 2.0 m to potential temperature. Recalculate the descriptive statistics for the potential temperatures at 0.4 and 2m; how does this adjustment impact the values?

In [20]:
DALR <- -0.01 #Km^-1
z0.4 <- 0.4 #m
z2 <- 2 #m

df2 <- select(df, T_0.4m..K., T_2m..K.) |>
    mutate(theta0.4 = (T_0.4m..K. + DALR*z0.4), theta2 = (T_2m..K. + DALR*z2))

summary(df2)

   T_0.4m..K.       T_2m..K.        theta0.4         theta2     
 Min.   :282.0   Min.   :284.0   Min.   :282.0   Min.   :284.0  
 1st Qu.:285.6   1st Qu.:287.5   1st Qu.:285.5   1st Qu.:287.4  
 Median :294.6   Median :294.0   Median :294.6   Median :294.0  
 Mean   :293.6   Mean   :293.2   Mean   :293.6   Mean   :293.1  
 3rd Qu.:301.0   3rd Qu.:299.0   3rd Qu.:301.0   3rd Qu.:299.0  
 Max.   :305.4   Max.   :301.4   Max.   :305.4   Max.   :301.4  

In [ ]:
The adjustment 

## Question 3